universe/syncer.go

```diff
@@ -52,6 +61,23 @@ func NewSimpleSyncer(cfg SimpleSyncCfg) *SimpleSyncer {
 func (s *SimpleSyncer) executeSync(ctx context.Context, diffEngine DiffEngine,
 	syncType SyncType, idsToSync []Identifier) ([]AssetSyncDiff, error) {
 
+	// Prevent the syncer from running twice. We also don't want to block on
+	// acquiring the mutex, so we use a secondary flag that is guarded by a
+	// mutex instead.
+	s.isSyncingMtx.Lock()
+	if s.isSyncing {
```

Comment:
When would this be called concurrently? Just users trying to sync multiple times on the CLI/RPC?

Reply:
This is to avoid the timed universe sync to run while we're still syncing from an RPC sync request. I noticed this while running the hour-long 10k mint where we spin up a secondary node and sync for almost 40 minutes, then the timed sync actually caused the whole test to fail because it interfered with the already running sync.

Reply:
Feels like this construct could be replaced with an `atomic.Bool`?

Reply:
Ah yes, I wasn't aware of the `CompareAndSwap` method, which does exactly what we want.